lib/TabTracker.js

```diff
@@ -0,0 +1,89 @@
+const tabs = require("sdk/tabs");
+const ss = require("sdk/simple-storage");
+
+const NEWTAB_URL = "resource://activity-streams/data/content/activity-streams.html#/";
+
+const TabTracker = {
+  _openTabs: [],
+
+  init: function() {
+    this.onOpen = this.onOpen.bind(this);
+    tabs.on("open", this.onOpen);
+    if (!ss.storage.tabData) {
+      ss.storage.tabData = {};
+    }
+  },
+
+  uninit: function() {
+    for (let i in this._openTabs) {
+      let tab = this._openTabs[i];
+      tab.removeListener("pageshow", this.logShow);
```

Comment:
The listener won't actually be removed. This is a bug elsewhere in the code as well.

if a listener was registered with e.g. `this.logShow.bind(this)`, when `removeListener` is invoked, `this.logShow === this.logShow.bind(this) === false `, i.e. it won't get removed